execute/table.go

```diff
@@ -107,119 +115,223 @@ func AddNewCols(t flux.Table, builder TableBuilder) ([]int, error) {
 // This function assumes builder and t have the same column schema.
-func AppendTable(t flux.Table, builder TableBuilder) {
+func AppendTable(t flux.Table, builder TableBuilder) error {
 	if len(t.Cols()) == 0 {
-		return
+		return nil
 	}
 
-	t.Do(func(cr flux.ColReader) error {
-		AppendCols(cr, builder)
-		return nil
+	return t.Do(func(cr flux.ColReader) error {
+		return AppendCols(cr, builder)
 	})
 }
 
 // AppendMappedCols appends all columns from cr onto builder.
 // The colMap is a map of builder column index to cr column index.
-func AppendMappedCols(cr flux.ColReader, builder TableBuilder, colMap []int) {
+func AppendMappedCols(cr flux.ColReader, builder TableBuilder, colMap []int) error {
 	for j := range builder.Cols() {
-		AppendCol(j, colMap[j], cr, builder)
+		if err := AppendCol(j, colMap[j], cr, builder); err != nil {
+			return err
+		}
 	}
+	return nil
 }
 
 // AppendCols appends all columns from cr onto builder.
 // This function assumes that builder and cr have the same column schema.
-func AppendCols(cr flux.ColReader, builder TableBuilder) {
+func AppendCols(cr flux.ColReader, builder TableBuilder) error {
 	for j := range builder.Cols() {
-		AppendCol(j, j, cr, builder)
+		if err := AppendCol(j, j, cr, builder); err != nil {
+			return err
+		}
 	}
+	return nil
 }
 
 // AppendCol append a column from cr onto builder
 // The indexes bj and cj are builder and col reader indexes respectively.
-func AppendCol(bj, cj int, cr flux.ColReader, builder TableBuilder) {
+func AppendCol(bj, cj int, cr flux.ColReader, builder TableBuilder) error {
+	if cj < 0 || cj > len(cr.Cols()) {
+		return errors.New("AppendCol column reader index out of bounds")
+	}
+	if bj < 0 || bj > len(builder.Cols()) {
+		return errors.New("AppendCol builder index out of bounds")
+	}
 	c := cr.Cols()[cj]
+
 	switch c.Type {
 	case flux.TBool:
-		builder.AppendBools(bj, cr.Bools(cj))
+		return builder.AppendBools(bj, cr.Bools(cj))
 	case flux.TInt:
-		builder.AppendInts(bj, cr.Ints(cj))
+		return builder.AppendInts(bj, cr.Ints(cj))
 	case flux.TUInt:
-		builder.AppendUInts(bj, cr.UInts(cj))
+		return builder.AppendUInts(bj, cr.UInts(cj))
 	case flux.TFloat:
-		builder.AppendFloats(bj, cr.Floats(cj))
+		return builder.AppendFloats(bj, cr.Floats(cj))
 	case flux.TString:
-		builder.AppendStrings(bj, cr.Strings(cj))
+		return builder.AppendStrings(bj, cr.Strings(cj))
 	case flux.TTime:
-		builder.AppendTimes(bj, cr.Times(cj))
+		return builder.AppendTimes(bj, cr.Times(cj))
 	default:
 		PanicUnknownType(c.Type)
 	}
+	return nil
 }
 
 // AppendRecord appends the record from cr onto builder assuming matching columns.
-func AppendRecord(i int, cr flux.ColReader, builder TableBuilder) {
+func AppendRecord(i int, cr flux.ColReader, builder TableBuilder) error {
+
+	if !ColsMatch(builder, cr) {
+		return errors.New("AppendRecord column schema mismatch")
+	}
+	for j, c := range builder.Cols() {
+		var err error
+		switch c.Type {
```

Comment:
There's a utility for this now in `ValueForRow`. I think that's what it's called. It already existed, but I modified it slightly and started using it in a recent commit.